### PR TITLE
chore: remove "using namespace " from AZCore Memory

### PR DESCRIPTION
**Pull request by pollend**
REF: https://github.com/o3de/o3de/issues/6281

Signed-off-by: Michael Pollind <mpollind@gmail.com>

### CODE REVIEW

File: Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp [hunk 1]
```diff
@@ -16,8 +16,11 @@
 
 #include <AzCore/Debug/StackTracer.h>
 
-using namespace AZ;
-using namespace AZ::Debug;
+namespace AZ 
+{
+
+namespace Debug
+{
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp:23]: Nit: the formatting here doesn't match the expected one ( namespace contents are indented )
If possible, I'd suggest using c++17 nested namespace feature : 
```
namespace AZ::Debug 
{

}
```

File: Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp [hunk 2]
```diff
@@ -9,7 +9,8 @@
 #include <AzCore/Memory/Memory.h>
 #include <AzCore/Memory/AllocatorManager.h>
 
-using namespace AZ;
+namespace AZ
+{
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp:13]: Missing indent for the namespace contents.

File: Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapAllocator.cpp [hunk 3]
```diff
@@ -13,7 +13,8 @@
 
 #include <AzCore/std/functional.h>
 
-using namespace AZ;
+namespace AZ 
+{
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapAllocator.cpp:18]: Missing indent on namespace contents.

File: Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapSchema.cpp [hunk 4]
```diff
@@ -9,8 +9,9 @@
 #include <AzCore/Memory/BestFitExternalMapSchema.h>
 #include <AzCore/Memory/SystemAllocator.h>
 
-using namespace AZ;
-
+namespace AZ
+{
+
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapSchema.cpp:14]: same as above

File: Code/Framework/AzCore/AzCore/Memory/PoolSchema.cpp [hunk 5]
```diff
@@ -325,8 +325,8 @@ namespace AZ
     };
 }
 
-using namespace AZ;
-
+namespace AZ
+{
 //=========================================================================
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/PoolSchema.cpp:330]: same as above

File: Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp [hunk 6]
```diff
@@ -36,8 +36,8 @@
 #endif
 
 
-using namespace AZ;
-
+namespace AZ
+{
 //////////////////////////////////////////////////////////////////////////
```
Comment (nemerle) [inline Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp:41]: same as above + this file is missing new-line at the end

File: Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp (hunk 2 above)
Comment (pollend) [reply Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp:13]: umm didn't want to modify the entire file but that can be done

**Review by lumberyard-employee-dm — APPROVED** [review]
FYI @amznestebanpapp is making changes in this code area in PR #6390

File: Code/Framework/AzCore/AzCore/Memory/PoolSchema.h [hunk 7]
```diff
@@ -5,8 +5,7 @@
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  *
  */
-#ifndef AZ_POOL_ALLOCATION_SCHEME_H
-#define AZ_POOL_ALLOCATION_SCHEME_H
+#pragma once
```
Comment (amznestebanpapp) [inline Code/Framework/AzCore/AzCore/Memory/PoolSchema.h:8]: this file is missing the removal of:
```
#endif // AZ_POOL_ALLOCATION_SCHEME_H
#pragma once
```
From the end of the file

File: Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp [hunk 8]
```diff
@@ -21,26 +21,25 @@
...
+#include <AzCore/Memory/HphaSchema.h>
 #elif AZCORE_SYSTEM_ALLOCATOR == AZCORE_SYSTEM_ALLOCATOR_MALLOC
-    #include <AzCore/Memory/MallocSchema.h>
+#include <AzCore/Memory/MallocSchema.h>
 #elif AZCORE_SYSTEM_ALLOCATOR == AZCORE_SYSTEM_ALLOCATOR_HEAP
-    #include <AzCore/Memory/HeapSchema.h>
+#include <AzCore/Memory/HeapSchema.h>
 #else
-    #error "Invalid allocator selected for SystemAllocator"
+#error "Invalid allocator selected for SystemAllocator"
```
Comment (amznestebanpapp) [inline Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp:35]: these indentations were done on purpose to make it easier to read

**Review by amznestebanpapp — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp [hunk 9]
```diff
@@ -16,506 +16,521 @@
...
+            AZ_Assert(iterBool.second, "Memory address 0x%p is already allocated and in the records!", address);
+        }
+
+        Debug::AllocationInfo& ai = iterBool.first->second;
+        ai.m_byteSize = byteSize;
+        ai.m_alignment = static_cast<unsigned int>(alignment);
+        if ((m_saveNames || m_mode == RECORD_FULL) && name && fileName)
+        {
+            // In RECORD_FULL mode or when specifically enabled in app descriptor with
+        // In RECORD_FULL mode or when specifically enabled in app descriptor with
```
Comment (hultonha) [inline Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp:147]: It looks like there's some duplicated lines here that were accidentally added

File: Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp [hunk 10]
```diff
@@ -16,506 +16,521 @@
...
-        // \todo store the load factor at the last rehash to avoid unnecessary rehash
-        if (m_records.load_factor() < 0.9f)
+        AllocationInfo allocationInfo;
         {
-            m_records.rehash(0);
+            AZStd::scoped_lock lock(m_recordsMutex);
+            Debug::AllocationRecordsType::iterator iter = m_records.find(address);
+            // We cannot assert if an allocation does not exist because allocations may have been made before tracking was enabled.
+        // We cannot assert if an allocation does not exist because allocations may have been made before tracking was enabled. 
+            // We cannot assert if an allocation does not exist because allocations may have been made before tracking was enabled.
```
Comment (hultonha) [inline Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp:258]: nit: Same here? Two identical lines and the indentation is off on one of them

File: Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp (hunk 2 above)
Comment (hultonha) [reply Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp:13]: I've reviewing with white space changes off and it's not too bad 👍

File: Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapSchema.cpp [hunk 11]
```diff
@@ -9,194 +9,201 @@
...
+                    size_t newBlockSize = curBlock->first + nextBlock->first;
+                    char* newBlockAddress = curStart;
+                    m_freeChunksMap.erase(nextBlock);
+                    FreeMapType::iterator toErase = curBlock;
+                    ++curBlock;
+                    m_freeChunksMap.erase(toErase);
+                    FreeMapType::iterator newBlock = m_freeChunksMap.insert(AZStd::make_pair(newBlockSize, newBlockAddress)).first;
+                    if (curBlock != m_freeChunksMap.end() &&
+                        newBlockSize <
+                            curBlock->first) // if the newBlock in before the next in the list, update next in the list to current
```
Comment (hultonha) [inline Code/Framework/AzCore/AzCore/Memory/BestFitExternalMapSchema.cpp:193]: suggestion: Could we move the comment to above the if and then this line won't be formatted so strangely 😝

File: Code/Framework/AzCore/AzCore/Memory/PoolSchema.cpp [hunk 12]
```diff
@@ -315,1090 +321,1090 @@ namespace AZ
...
+        // External requests for garbage collection may come from any thread, and the
+        // garbage collection operation isn't threadsafe, which can lead to crashes.
+        //
+        // Due to the low memory consumption of this allocator in practice on Dragonfly
+        // (~3kb) it makes sense to not bother with garbage collection and leave it to
+        // occur exclusively in the destruction of the allocator.
+        //
+        // TODO: A better solution needs to be found for integrating back into mainline
+    // TODO: A better solution needs to be found for integrating back into mainline 
+        // TODO: A better solution needs to be found for integrating back into mainline
```
Comment (hultonha) [inline Code/Framework/AzCore/AzCore/Memory/PoolSchema.cpp:721]: nit: duplicate comments again 🤔

**Review by hultonha — COMMENTED** [review]
Overall looks good just have a few comments, once they are addressed give me a poke and I can approve 👍

**Pull request merged**